eventsource-client/src/client.rs

```diff
@@ -343,3 +329,28 @@ where
         }
     }
 }
+
+fn delay(dur: Duration, description: &str) -> Sleep {
```

Comment:
Rearranging some stuff so that `ReconnectingRequest` implementation can be grouped together (just for readability.)